RiotSwiftUI/Modules/Settings/Notifications/Service/MatrixSDK/MXNotificationSettingsService.swift

```diff
@@ -71,16 +70,54 @@ class MXNotificationSettingsService: NotificationSettingsServiceType {
         session.notificationCenter.removeRule(rule)
     }
     
-    func updatePushRuleActions(for ruleId: String, enabled: Bool, actions: NotificationActions?) {
-        guard let rule = session.notificationCenter.rule(byId: ruleId) else { return }
-        session.notificationCenter.enableRule(rule, isEnabled: enabled)
+    func updatePushRuleActions(for ruleId: String,
+                               enabled: Bool,
+                               actions: NotificationActions?,
+                               completion: ((Result<Void, Error>) -> Void)?) {
         
-        if let actions = actions {
-            session.notificationCenter.updatePushRuleActions(ruleId,
-                                                             kind: rule.kind,
-                                                             notify: actions.notify,
-                                                             soundName: actions.sound,
-                                                             highlight: actions.highlight)
+        guard let rule = session.notificationCenter.rule(byId: ruleId) else {
+            completion?(.success)
+            return
+        }
+        
+        guard let actions = actions else {
+            enableRule(rule: rule, enabled: enabled, completion: completion)
+            return
+        }
+        
+        // Updating the actions before enabling the rule allows the homeserver to triggers just one sync update
+        session.notificationCenter.updatePushRuleActions(ruleId,
+                                                         kind: rule.kind,
+                                                         notify: actions.notify,
+                                                         soundName: actions.sound,
+                                                         highlight: actions.highlight) { [weak self] error in
+            switch error.result {
+            case .success:
+                self?.enableRule(rule: rule, enabled: enabled, completion: completion)
+            case .failure:
+                completion?(error.result)
+            }
         }
     }
 }
+
+private extension MXNotificationSettingsService {
+    func enableRule(rule: MXPushRule, enabled: Bool, completion: ((Result<Void, Error>) -> Void)?) {
+        session.notificationCenter.enableRule(rule, isEnabled: enabled) { error in
+            completion?(error.result)
```

Comment:
Hm this looks a bit confusing. If I understand correctly `error.result` will either give you error or success, but the reader would not know this without looking at the implementation of the `result` extension. Being explicit via some `if let/else` may mean more lines of code but is in my opinion clearer.

Reply:
Yeah it's just converting an objective-c style `Error?` to a swifty `Result<Void, Error>`.
I'll prefer to keep the extension because I need it several times.

Reply:
Isn't it just this?
```objc
if let error = error {
    completion(.failure(error)
} else {
    completion(.success(()))
}
```

You do use `error.result` several times but with similar level of ambiguity, see this:
```objc
switch error.result {
case .success:
  self?.enableRule(rule: rule, enabled: enabled, completion: completion)
case .failure:
  completion?(error.result)
}
```

This is even longer than doing the `if/else` and we have to unpack `error.result` twice so it really looks like the wrong kind of API to me